pkg/kubelet/rkt/rkt.go

```diff
@@ -1547,8 +1715,22 @@ func (r *Runtime) GarbageCollect(gcPolicy kubecontainer.ContainerGCPolicy) error
 		if strings.HasPrefix(serviceName, kubernetesUnitPrefix) {
 			rktUUID := getRktUUIDFromServiceFileName(serviceName)
 			if _, ok := allPods[rktUUID]; !ok {
+				serviceFile := serviceFilePath(f.Name())
+				id, name, namespace, err := podDetailsFromServiceFile(serviceFile)
+				if err == nil {
+					r.cleanupPodNetwork(&api.Pod{
```

Comment:
Everytime there is an error, we invoke cleanupPodNetwork. Is it possible that we defer a validation function? In this function, check if the current operation succeeded, and then invoke cleanupPodNetwork if needed.